File: java_solutions/symmetric_tree.java
public class SymmetricTree {
    public class TreeNode {
      int val;
      TreeNode left;
      TreeNode right;
      TreeNode() {}
      TreeNode(int val) { this.val = val; }
      TreeNode(int val, TreeNode left, TreeNode right) {
          this.val = val;
          this.left = left;
          this.right = right;
      }
  }
    class Solution {
        public boolean dfs(TreeNode leftchild, TreeNode rightchild){
            if (leftchild == null && rightchild == null)
                return true;
            if (leftchild == null || rightchild == null)
                return false;
            if(leftchild.val != rightchild.val)
                return false;
            return (dfs(leftchild.left, rightchild.right) &&
                    dfs(leftchild.right, rightchild.left));
        }

        public boolean isSymmetric(TreeNode root) {
            return dfs(root.left, root.right);

        }
    }

  
}

File: python_solutions/symmetric_tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        def dfs(leftChild, rightChild):
            if not leftChild and not rightChild:
                return True
            if not leftChild or not rightChild:
                return False
            if leftChild.val != rightChild.val:
                return False
            
            return (dfs(leftChild.left, rightChild.right) and 
                    dfs(leftChild.right, rightChild.left))
            
        return dfs(root.left, root.right)
        
